File: utils-debug/src/main/java/nl/ngti/androscope/service/AndroscopeServiceStatus.java
package nl.ngti.androscope.service;

import androidx.annotation.NonNull;

public final class AndroscopeServiceStatus {

    private static final int STATUS_OK = 0;
    private static final int STATUS_FAILED = 1;
    private static final int STATUS_STOPPED = 2;

    private final String mMessage;
    private final int mStatus;

    private AndroscopeServiceStatus(String message, int status) {
        mMessage = message;
        mStatus = status;
    }

    @NonNull
    static AndroscopeServiceStatus success(String message) {
        return new AndroscopeServiceStatus(message, STATUS_OK);
    }

    @NonNull
    static AndroscopeServiceStatus error(String message) {
        return new AndroscopeServiceStatus(message, STATUS_FAILED);
    }

    @NonNull
    static AndroscopeServiceStatus stopped(String message) {
        return new AndroscopeServiceStatus(message, STATUS_STOPPED);
    }

    public String getMessage() {
        return mMessage;
    }

    public boolean isRestartNeeded() {
        return mStatus != STATUS_OK;
    }
}

File: lib-androscope/src/main/java/nl/ngti/androscope/service/AndroscopeServiceStatus.kt
package nl.ngti.androscope.service

internal class AndroscopeServiceStatus private constructor(
        val message: String,
        private val status: Int
) {

    val isRestartNeeded: Boolean
        get() = status != STATUS_OK

    companion object {
        private const val STATUS_OK = 0
        private const val STATUS_FAILED = 1
        private const val STATUS_STOPPED = 2

        fun success(message: String): AndroscopeServiceStatus {
            return AndroscopeServiceStatus(message, STATUS_OK)
        }

        fun error(message: String): AndroscopeServiceStatus {
            return AndroscopeServiceStatus(message, STATUS_FAILED)
        }

        fun stopped(message: String): AndroscopeServiceStatus {
            return AndroscopeServiceStatus(message, STATUS_STOPPED)
        }
    }

}
